picobot/video_editor.py

```diff
@@ -0,0 +1,46 @@
+from pathlib import Path
+import ffmpeg
+
+
+def sticker_from_video(mp4_path: Path):
+    '''
+    Converts the given video to a proper format that can be uploaded as a sticker.
+    Telegram accepts WEBM VP9 with a maximum size of 512x512 pixels with maximum 3 seconds duration.
+    '''
+    probe = ffmpeg.probe(mp4_path)
+    video_stream = next(
+        (stream for stream in probe['streams'] if stream['codec_type'] == 'video'), None
+    )
+    width = int(video_stream['width'])
+    height = int(video_stream['height'])
+    duration = float(probe['format']['duration'])
+
+    check_duration(duration)
+
+    (width, height) = estimate_video_sticker_size(width, height)
+    vid_output_path = mp4_path.with_suffix('.webm')
+    (
+        ffmpeg.input(mp4_path)
+        .filter('scale', width, height)
+        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
+        .run()
+    )
+    return vid_output_path
+
+
+def check_duration(duration: float):
+    if duration > 3.0:
+        raise Exception('video duration exceeds limits')
+
+
+def estimate_video_sticker_size(width: int, height: int):
```

Comment:
[3] Annotate return types as well:

```suggestion
def estimate_video_sticker_size(width: int, height: int) -> tuple[int, int]:
```